### PR TITLE
docs: auto-generate API docs

### PR DESCRIPTION
**Pull request by e111077**
Updates API docs in our markdown files with Lit Analyzer by manually running `npm run update-docs`

### CODE REVIEW

File: docs/components/tabs.md [hunk 1]
```diff
@@ -369,3 +369,96 @@ Token                                       | Default value
...
+
+<!-- mdformat on(autogenerated might break rendering in catalog) -->
+
+#### Methods
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Method | Parameters | Returns | Description
+--- | --- | --- | ---
+`[ANIMATE_INDICATOR]` | `previousTab` | `void` |
```
Comment (AndrewJakubowicz) [inline docs/components/tabs.md:435]: This public method uses a symbol to obfuscate it from public use. Should probably be omitted from public docs.

File: docs/components/tabs.md [hunk 2]
```diff
@@ -369,3 +369,96 @@ Token                                       | Default value
...
```
Comment (AndrewJakubowicz) [inline docs/components/tabs.md:460]: Nit: Remove symbol from public API.

File: docs/components/chip.md [hunk 3]
```diff
@@ -480,3 +480,96 @@ Token                                   | Default value
...
+
+### MdChipSet
+
+#### Properties
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Property | Attribute | Type | Default | Description
+--- | --- | --- | --- | ---
+`chips` |  | `Chip[]` | `undefined` |
```
[the remark this answers is not in the document]
Comment (e111077) [reply docs/components/chip.md:497]: Analyzer will output readonly but i'm getting worried that some of these property tables are getting mighty wide. Can we consider this later?

File: docs/components/dialog.md [hunk 4]
```diff
@@ -238,3 +238,51 @@ Token                               | Default value
...
+
+<!-- mdformat on(autogenerated might break rendering in catalog) -->
+
+#### Events
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Event | Type | Description
+--- | --- | ---
+`open` | `undefined` | Dispatched when the dialog is opening before any animations.
```
[the remark this answers is not in the document]
Comment (e111077) [reply docs/components/dialog.md:280]: we haven't been up to date with setting `@event open {CustomEvent<Asdf>} description of stuff` i will remove it for now

File: docs/components/fab.md [hunk 5]
```diff
@@ -472,3 +472,62 @@ Add with a different theme applied](images/fab/theming-branded.webp)
...
+<!-- mdformat on(autogenerated might break rendering in catalog) -->
+
+#### Methods
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Method | Parameters | Returns | Description
+--- | --- | --- | ---
+`getRenderClasses` | _None_ | `{ primary: boolean; secondary: boolean; tertiary: boolean; small: boolean; lowered: boolean; large: boolean; extended: boolean; }` | 
+`getRenderClasses` | _None_ | `{ primary: boolean; secondary: boolean; tertiary: boolean; lowered: boolean; small: boolean; large: boolean; extended: boolean; }` |
```
[the remark this answers is not in the document]
Comment (e111077) [reply docs/components/fab.md:529]: I need to handle overrides despite this should be protected

File: docs/components/fab.md [hunk 6]
```diff
@@ -472,3 +472,62 @@ Add with a different theme applied](images/fab/theming-branded.webp)
...
+
+<!-- mdformat on(autogenerated might break rendering in catalog) -->
+
+#### Methods
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Method | Parameters | Returns | Description
+--- | --- | --- | ---
+`getRenderClasses` | _None_ | `{ primary: boolean; secondary: boolean; tertiary: boolean; lowered: boolean; small: boolean; large: boolean; extended: boolean; }` |
```
[the remark this answers is not in the document]
Comment (e111077) [reply docs/components/fab.md:502]: will do in another CL

File: docs/components/button.md [hunk 7]
```diff
@@ -698,3 +698,110 @@ Token                               | Default value
...
+
+#### Properties
+
+<!-- mdformat off(autogenerated might break rendering in catalog) -->
+
+Property | Attribute | Type | Default | Description
+--- | --- | --- | --- | ---
+`disabled` | `disabled` | `boolean` | `false` | Whether or not the button is disabled.
+`href` | `href` | `string` | `''` | The URL that the link button points to.
+`target` | `target` | `string` | `''` | Where to display the linked `href` URL for a link button. Common options<br>include `_blank` to open in a new tab.
```
[the remark this answers is not in the document]
Comment (e111077) [reply docs/components/button.md:717]: it looks great, and currently we do not have code in our jsdoc so we fine

**Comment by e111077** [comment]
addressed comments in CL

**Comment by e111077** [comment]
Copybara merged here: https://github.com/material-components/material-web/commit/9f3e55d79a2da1af0e6b41ef479e84a9fb106c63

**Pull request closed without merging**